src/types.rs

```diff
@@ -124,19 +123,19 @@ impl<'a> SegmentParam<'a> {
     }
 }
 
-impl<'a> fmt::Display for SegmentParam<'a> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            SegmentParam::LifeTime(ref lt) => {
-                write!(f, "{}", pprust::lifetime_to_string(lt))
-            }
-            SegmentParam::Type(ref ty) => {
-                write!(f, "{}", pprust::ty_to_string(ty))
-            }
-            SegmentParam::Binding(ref binding) => {
-                write!(f, "{} = {}", binding.ident, pprust::ty_to_string(&*binding.ty))
-            }
-        }
+impl<'a> Rewrite for SegmentParam<'a> {
+    fn rewrite(&self, context: &RewriteContext, width: usize, offset: usize) -> Option<String> {
```

Comment:
This doesn't uphold the variant that the single-line result of a rewrite does not exceed `width` characters, which could lead to Bad Stuff..